### PR TITLE
Executors docker edition

### PR DESCRIPTION
**Pull request by eseliger**
This PR adds a docker version of executors.

Next up: 
- Making it work for SSBC by allowing custom URLs for the SRC_ENDPOINT variable (this variable I want to get rid off soon, but otherwise I can't test it)
- Craft a k8s manifest to run in locally in the docker managed k8s to verify it can work
- Create a docker-compose extension that can run this successfully to verify it can also work for compose deployments

## Test plan

Ran a main-dry-run and a executor-patch-notest build to validate they pass.

### CODE REVIEW

File: enterprise/cmd/frontend/internal/executorqueue/queues/batches/transform.go [hunk 1]
```diff
@@ -81,7 +81,9 @@ func transformRecord(ctx context.Context, s BatchesStore, job *btypes.BatchSpecW
 
 	frontendURL := conf.Get().ExternalURL
 
-	srcEndpoint, err := makeURL(frontendURL, accessToken)
+	// TODO: This must be configurable and not just be the external URL.
+	// For OOTB deployments we want to hit the frontend internally.
+	srcEndpoint, err := makeURL("http://host.docker.internal:3081", accessToken)
```
Comment (eseliger) [inline enterprise/cmd/frontend/internal/executorqueue/queues/batches/transform.go:86]: This is a blocker rn!

File: enterprise/cmd/executor/vm-image/executor.json [hunk 2]
```diff
@@ -37,8 +37,8 @@
       "access_key": "{{user `awsAccessKey`}}",
       "secret_key": "{{user `awsSecretKey`}}",
       "aws_polling": {
-        "delay_seconds": "{{user `awsMaxAttempts`}}",
-        "max_attempts": "{{user `awsPollDelaySeconds`}}"
+        "delay_seconds": "{{user `awsPollDelaySeconds`}}",
+        "max_attempts": "{{user `awsMaxAttempts`}}"
```
Comment (eseliger) [inline enterprise/cmd/executor/vm-image/executor.json:41]: This was quite a performance boost for CI :lolsob:

File: enterprise/cmd/executor/hash.sh [hunk 3]
```diff
@@ -9,6 +9,7 @@ cleanup() {
 trap cleanup EXIT
 
 # Do not embed build flags to produce a stable output
+# TODO: We do need to consider the src-cli version here though, otherwise changing that constant doesn't have any effect here.
```
Comment (eseliger) [inline enterprise/cmd/executor/hash.sh:12]: TODO

**Review by efritz — APPROVED** [review]

File: enterprise/cmd/executor/vm-image/executor.json (hunk 2 above)
Comment (efritz) [reply enterprise/cmd/executor/vm-image/executor.json:41]: Oh no 🤦

**Pull request merged**